### ISSUE TITLE
"Should maximize" and layouts in beta 7.0.0

### ISSUE THREAD

**Issue by hilfazer** [bitwes/Gut#181]
Gut control doesn't maximize if "should maximize" is checked.

Setting Gut control's layout to Full Rect will make it take all available space on launch but it won't be resizing together with application's window:
![Gut_700_fullRectLayout](https://user-images.githubusercontent.com/29497869/85051600-0083bb80-b198-11ea-86af-a43b5f0a2ed3.png)
It does resize in 6.8.3.

**Comment by bitwes** [comment]
Due to changes, I had to add in a `call_deferred` for the maximize to take effect.  Fixed another issue where un-maximizing would always reposition GUI to `[0, 0]`.  These changes have been pushed up.

The Layout->Full Rect issue is more difficult.  The short story is that GUT fakes the control in the editor.  A different instance of the GUI is added when the scene runs.  It copies the size, but not layout or any other properties.  Since maximize is fixed, this is less of an issue.

The long version of the story is real long and has to do with trying to create a pseudo autoload instance of the `addons/gut/utils.gd`.  This is the reason for the change to the `gut_ready` signal instead of being able to use `_ready`.  This approach worked fine for the command line but broke the scene approach causing a bunch of hack workarounds.
